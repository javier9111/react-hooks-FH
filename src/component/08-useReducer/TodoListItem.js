import React from "react";

export const TodoListItem = ({ todo, handleToggle, handleDelete, index }) => {
  return (
    <li key={todo.id} className="list-group-item">
      <p
        onClick={() => handleToggle(todo.id)}
        className={`text-center ${todo.done && "complete"}`}
      >
        {index + 1}.{todo.desc}
      </p>
      <button onClick={() => handleDelete(todo.id)} className="btn btn-danger">
        borrar
      </button>
    </li>
  );
};
