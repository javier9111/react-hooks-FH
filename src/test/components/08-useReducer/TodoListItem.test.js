import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { TodoListItem } from "../../../component/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Test on <TodoListItem/>", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
      index={0}
    />
  );
  test("should show correctly", () => {
    //  console.log(demoTodos[0]);
    expect(wrapper).toMatchSnapshot();
  });

  test("should call the delete function", () => {
    wrapper.find(".btn-danger").simulate("click");
    expect(handleDelete).toHaveBeenCalled();
  });

  test("should call the toggle function", () => {
    wrapper.find("p").simulate("click");
    expect(handleToggle).toHaveBeenCalled();
  });

  test("should show the text correctly", () => {
    //the paragraf text
    const todo = demoTodos[0];
    expect(wrapper.find("p").text().trim()).toBe(`1.${todo.desc}`);
  });
  test("should contain the complete class", () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        handleToggle={handleToggle}
        handleDelete={handleDelete}
        index={0}
      />
    );

    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
});
