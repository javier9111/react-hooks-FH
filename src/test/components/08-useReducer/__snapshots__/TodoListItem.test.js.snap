// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoListItem/> should show correctly 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="text-center false"
    onClick={[Function]}
  >
    1
    .
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    borrar
  </button>
</li>
`;
